### DISCUSSION TITLE
How to add URL Dependencies dynamically?

### DISCUSSION THREAD

**Discussion by jahugg** [parcel-bundler/parcel#7301] [Q&A]
### Question
How can I add URL dependencies via JavaScript using **variables** instead of **string literals** as first parameter in the URL constructor?

works:
```
let url = new URL( "myimage.jpg", import.meta.url);
```

does not work:
```
let path = "myimage.jpg";
let url = new URL( path, import.meta.url);
```

### Docs
The [docs](https://parceljs.org/languages/javascript/#url-dependencies) clearly state you have to provide a **string literal** as first parameter of the URL constructor. Otherwise it will not be recognized as a dependency.
Adding them manually one by one is not an option in my scenario but I'm happy to receive suggestions on how to do this differently.

### Context
I'm creating an app that uses the resolver-glob plugin to import the images within a given directory as a JavaScript object. I then iterate over the object to dynamically add the images to the app.

**Comment by Rock-n-Roll-CRC** [comment]
I know that the question is sort of old, but have you figured out how to implement this? I'm stuck with the same problem as you did.
